platform_plugin_turnitin/api/v1/views.py

```diff
@@ -0,0 +1,420 @@
+        upload_turnitin_submission_file():
+            Handle the upload of the user's file to Turnitin.
+        create_turnitin_submission_object():
+            Create a Turnitin submission object based on the user's data.
+        get_submission_status(submission_id: str):
+            Retrieve the status of the latest Turnitin submission for the user.
+        generate_similarity_report(submission_id: str):
+            Initialize the generation of a similarity report for the user's latest Turnitin submission.
+        get_similarity_report_status(submission_id: str):
+            Retrieve the status of the similarity report for the user's latest Turnitin submission.
+        create_similarity_viewer(submission_id: str):
+            Create a Turnitin similarity viewer for the user's latest submission.
+    """
+
+    def __init__(self, request, user, course) -> None:
+        self.request = request
+        self.user = user
+        self.course = course
+        self.first_name, self.last_name = get_fullname(self.user.profile.name)
+
+    def accept_eula_agreement(self) -> RequestsResponse:
+        """
+        Submit acceptance of the EULA for the current user.
+
+        Args:
+            user_data (dict): A dictionary containing user data
+
+        Returns:
+            RequestsResponse: The response after accepting the EULA.
+        """
+        payload = {
+            "user_id": str(self.user.id),
+            "accepted_timestamp": get_current_datetime(),
+            "language": "en-US",
+        }
+        return post_accept_eula_version(payload)
+
+    def upload_turnitin_submission_file(self) -> Response:
+        """
+        Handle the upload of the user's file to Turnitin.
+
+        Returns:
+            Response: The response after uploading the file to Turnitin.
+        """
+        turnitin_submission = self.create_turnitin_submission_object()
+
+        if turnitin_submission.status_code == status.HTTP_201_CREATED:
+            turnitin_submission_id = turnitin_submission.json()["id"]
+            submission = TurnitinSubmission(
+                user=self.user, turnitin_submission_id=turnitin_submission_id
+            )
+            submission.save()
+            uploaded_file = self.request.FILES["file"]
+            return Response(
+                put_upload_submission_file_content(
+                    turnitin_submission_id, uploaded_file
+                ).json()
+            )
+
+        return Response(turnitin_submission.json())
+
+    def create_turnitin_submission_object(self) -> RequestsResponse:
+        """
+        Create a Turnitin submission object based on the user's data.
+
+        Returns:
+            RequestsResponse: The response after creating the Turnitin submission object.
+        """
+        payload = {
+            "owner": self.user.id,
+            "title": f"{self.course}-{self.user.username}",
+            "submitter": self.user.id,
```

Comment:
does the submitter & owner have to be the user IDs? We usually use the username as the users identification

Reply:
This user ID is used when accepting the EULA, so it should be left like this